### PR TITLE
Add new meta

### PR DESCRIPTION
**Pull request by azoyan**

### CODE REVIEW

**Review by LouiseMedova — APPROVED** [review]

File: state/src/state.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+use gmeta::{metawasm, Metadata};
+use gstd::{prelude::*, ActorId};
+use onchain_nft_io::*;
+
+#[metawasm]
+pub trait Metawasm {
+    type State = <ContractMetadata as Metadata>::State;
+}
```
Comment (fluiderson) [inline state/src/state.rs:8]: You should copy here the logic from the state query enum from here:
https://github.com/gear-dapps/on-chain-nft/blob/472143a8475c629a120b678d5b4d3bd437c3b6ad/io/src/lib.rs#L26-L40

File: Cargo.toml [hunk 2]
```diff
@@ -9,13 +9,24 @@ gstd = { git = "https://github.com/gear-tech/gear.git", branch = "stable" }
...
+gear-lib = { git = "https://github.com/gear-dapps/gear-lib.git", tag = "0.3.4" }
+gear-lib-derive = { git = "https://github.com/gear-dapps/gear-lib.git", tag = "0.3.4" }
 hashbrown = "0.13.1"
+onchain-nft-io = { path = "io" }
 
 [dev-dependencies]
 gstd = { git = "https://github.com/gear-tech/gear.git", features = ["debug"], branch = "stable" }
 gtest = { git = "https://github.com/gear-tech/gear.git", branch = "stable" }
+gclient = { git = "https://github.com/gear-tech/gear.git" }
+onchain-nft-state = { path = "state", features = ["binary-vendor"] }
```
Comment (fluiderson) [inline Cargo.toml:21]:
```suggestion
```
Seems unused.

**Review by fluiderson — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (azoyan) [reply Cargo.toml:22]: used for gclient test

File: state/src/state.rs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        let metadata = if let Some((_id, metadata)) = state
+            .token
+            .token_metadata_by_id
+            .iter()
+            .find(|(id, _)| token_id.eq(id))
+        {
+            metadata.clone().unwrap_or_default()
+        } else {
+            TokenMetadata::default()
+        };
```
Comment (fluiderson) [inline state/src/state.rs:21]:
```suggestion
        let metadata = state
            .token
            .token_metadata_by_id
            .iter()
            .find(|(id, _)| token_id.eq(id))
            .and_then(|(_id, metadata)| metadata.clone())
            .unwrap_or_default();
```

File: state/src/state.rs [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            for (i, layer_item_id) in nft.iter().enumerate() {
+                if let Some((_id, layer_content)) =
+                    state.layers.iter().find(|(id, _)| (i as u128).eq(id))
+                {
+                    let s = layer_content
+                        .iter()
+                        .nth(*layer_item_id as usize)
+                        .expect("No such layer item");
+                    content.push(s.clone());
+                    return Some(TokenURI { metadata, content }.encode());
```
Comment (fluiderson) [inline state/src/state.rs:36]:
```suggestion
```
Returning on the first iteration? In the source function this statement was at the end of the block.

File: state/src/state.rs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                }
+            }
+        }
+
+        None
```
Comment (fluiderson) [inline state/src/state.rs:41]:
```suggestion
        Some(TokenURI { metadata, content }.encode())
```

**Review by fluiderson — CHANGES_REQUESTED** [review]

**Review by fluiderson — APPROVED** [review]

**Pull request merged**